executor.go

```diff
@@ -170,6 +172,15 @@ func (e *executor) run() {
 
 				switch f.runConfig.mode {
 				case defaultMode:
+					if e.distributedLocker != nil {
```

Comment:
This should be moved into the scheduler so that the lock and unlock and last for the interval. If the job is set to run once a minute and the schedulers start up at slightly different times, this implementation wouldn’t block the others unless the job took a long time to run. 

One way around it would be to force jobs using the distributed locker to use StartAt to make interval jobs deterministic.

Reply:
can't move it into the scheduler though because then we'd have to take into account all the different modes there. also, locking in the job continuous run function would likely result in a single scheduler running the same job over and over instead of other schedulers having a chance.